### PR TITLE
Xenophobic bugfixes

### PR DESCRIPTION
**Pull request by Morlic-fo**
The PR covers two fixes related to the xenophobic frenzy:
#### 1. Fixes issue #924

Bug was existing as population malus was proportional to current population. So negative populations received a "negative malus", i.e. a bonus.
Note that the formula is not changed but merely recasted to an actually readable form:

> Previous form of the formula: `Max(A,B)`
> with `B = Value - 3*size =: Value - b`,
> with `A = Value * (0.6 + (0.4 * (0.8 ^ n)))`
> 
> A is easily recasted to
> 
> ```
> A  =  Value *  [ (1-0.4) + 0.4 * 0.8^n] 
>    =  Value * (1 - 0.4 * (1 - 0.8^n))
>    =  Value - Value * 0.4*(1-0.8^n)
>    =: Value - a
> ```
> 
> It is then easy to see that `Max(A, B)` is equivalent to `Value - Min(a, b)`, i.e. what is used in the new code.
#### 2. Fixes Happiness malus:

Activation for the condition ignores planets with Concentration Camps. However, planets with concentration camps where still counted for the calculation of the happiness malus.

### CODE REVIEW

File: default/scripting/species/common/xenophobic.macros [hunk 1]
```diff
@@ -31,15 +31,16 @@ XENOPHOBIC_SELF
...
+                        PopulationCenter
+                        OwnedBy empire = Source.Owner
+                        Not OR [ 
+                            Species name = Source.Species
+                            Species name = "SP_EXOBOT"
+                        ]
+                        Not Population high = 0
+                        Not Contains Building name = "BLD_CONC_CAMP"
+                        WithinStarlaneJumps jumps = 5 condition = Source
                     ]
```
Comment (Morlic-fo) [inline default/scripting/species/common/xenophobic.macros:43]: Note that I changed the order of conditions to be consistent with the activation condition above.

**Comment by MatGB** [comment]
Right...

I was on entirely the wrong track looking for what was wrong with that code, thank you.

I do want to completely rework the way we do xenophobia at some point (so that it's always a per planet malus the way happiness already is), but having the existing maths actually work is a good thing. Thank you.

I won't be able to test this for the next 48hours or so but it ought to go into the release if possible, assuming it works feel free to merge in.

**Comment by Morlic-fo** [comment]
> I won't be able to test this for the next 48hours or so but it ought to go into the release if possible, assuming it works feel free to merge in.

Worked in my tests:
-Playing Trith and colonizing a hostile planet with some nearby natives gives 0 bonus/malus 
-Homeworld (with nearby natives) has same target population as before the changes.

**Comment by Vezzra** [comment]
Sounds reasonable. Took only a very brief look at the changes, seem to be ok too as far as I can tell.

Go ahead and merge. If you don't get around to it and no one raises any objections, I'll merge it before I create the release branch on Sunday (as you set the "Release v0.4.6" milestone).

**Comment by Vezzra** [comment]
Perfect, thanks!

**Pull request merged**